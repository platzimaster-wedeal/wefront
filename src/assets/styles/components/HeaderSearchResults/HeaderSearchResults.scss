@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../Breakpoints.scss';

.header-search-results {
  display: flex;
  flex-flow: column nowrap;
  width: 100%;
  position: relative;

  &__header {
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-around;

    @include media(tablet) {
      justify-content: flex-start;
    }
    & strong {
      @include font-size($font-size-normal);
      font-weight: 200;
      padding: 3px $space-size-normal;
      cursor: pointer;

      @include media(tablet) {
        padding: 3px $space-size-large;
      }
      
    }
    &--active {
      color: $color-primary;
      border-bottom: 3px solid $color-primary;
    }
  }

  &__results {
    padding: $space-size-normal 0;
  }
}
