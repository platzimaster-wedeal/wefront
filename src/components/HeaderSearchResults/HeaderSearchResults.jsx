import React, { useState } from "react";
import PhotoUser from "../../assets/static/img/users/user.jpg";

// Stlyes
import "../../assets/styles/Components/HeaderSearchResults/HeaderSearchResults.scss";

// Components
import ProfileSmallCard from "../ProfileSmallCard/ProfileSmallCard";
import PersonCard from "../PersonCard/PersonCard";

const HeaderSearchResults = ({ results = [] }) => {
 const [viewPersons, setViewPersons] = useState(true);

 const handleViewPersons = () => {
  setViewPersons(!viewPersons);
 };

 const RenderPersons = (hovered) => {
  if (hovered) {
   return (
    <PersonCard
     className="personCard"
     imgSource={PhotoUser}
     title="Tupac Shakur"
     subtitle="Musician">
     Live and dead for music.
    </PersonCard>
   );
  }

  return null;
 };

 const RenderProblems = (hovered) => {
  if (hovered) {
   return (
    <PersonCard
     className="personCard"
     imgSource={PhotoUser}
     title="Tupac Shakur"
     subtitle="Musician">
     Problem Status Card
    </PersonCard>
   );
  }

  return null;
 };

 return (
  <div className="header-search-results">
   <div className="header-search-results__header">
    <strong
     className={viewPersons && "header-search-results__header--active"}
     onClick={handleViewPersons}>
     Persons
    </strong>
    <strong
     className={!viewPersons && "header-search-results__header--active"}
     onClick={handleViewPersons}>
     Problems
    </strong>
   </div>
   <section className="header-search-results__results">
    {results && results.length > 0 ? (
     results.map((resp, i) => (
      <ProfileSmallCard
       key={i}
       imageSource={PhotoUser}
       altImage="Name"
       title="Tupac Shakur"
       subtitle="Musician">
       {(hovered) =>
        viewPersons ? RenderPersons(hovered) : RenderProblems(hovered)
       }
      </ProfileSmallCard>
     ))
    ) : (
     <span>Sorry, we can't found what you looking for</span>
    )}
   </section>
  </div>
 );
};

export default HeaderSearchResults;
