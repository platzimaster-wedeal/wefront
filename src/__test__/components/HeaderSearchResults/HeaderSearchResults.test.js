import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import HeaderSearchResults from "../../../components/HeaderSearchResults/HeaderSearchResults";

describe("Test for the component HeaderSearchResults :D", () => {
 const HeaderSearchResultsMounted = mount(<HeaderSearchResults />);
 const HeaderSearchResultsCreated = create(<HeaderSearchResults />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(HeaderSearchResultsCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(HeaderSearchResultsMounted.length).toEqual(1);
  });
  test("Should the tilte of first selected view of the component be a string", () => {
   expect(
    typeof HeaderSearchResultsMounted.find(
     ".header-search-results__header--active"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of first selected view of the component be Persons", () => {
   expect(
    HeaderSearchResultsMounted.find(
     ".header-search-results__header--active"
    ).text()
   ).toEqual("Persons");
  });
  test("Should the message of the component whene is not items to map be a string", () => {
   expect(
    typeof HeaderSearchResultsMounted.find(
     ".header-search-results__results span"
    ).text()
   ).toEqual("string");
  });
  test("Should the message of the component whene is not items to map be a Sorry, we can't found what you looking for", () => {
   expect(
    HeaderSearchResultsMounted.find(
     ".header-search-results__results span"
    ).text()
   ).toEqual("Sorry, we can't found what you looking for");
  });
 });
});
