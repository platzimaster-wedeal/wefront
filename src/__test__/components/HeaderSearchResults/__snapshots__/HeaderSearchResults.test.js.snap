// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component HeaderSearchResults :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="header-search-results"
>
  <div
    className="header-search-results__header"
  >
    <strong
      className="header-search-results__header--active"
      onClick={[Function]}
    >
      Persons
    </strong>
    <strong
      className={false}
      onClick={[Function]}
    >
      Problems
    </strong>
  </div>
  <section
    className="header-search-results__results"
  >
    <span>
      Sorry, we can't found what you looking for
    </span>
  </section>
</div>
`;
